ludwig/api.py

```diff
@@ -138,75 +138,91 @@ def train(
+            1 for validation, 2 for test), otherwise the dataset will be
+            randomly split.
+        :param training_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing training data.
+        :param validation_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing validation data.
+        :param test_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing test data.
+        :param training_set_metadata: (Union[str, dict], default: `None`)
+            metadata JSON file or loaded metadata.
+            Intermediate preprocess structure containing the mappings of the input
+            CSV created the first time a CSV file is used in the same
+            directory with the same name and a '.json' extension.
+        :param data_format: (str, default: `None`) format to interpret data sources. Will be inferred
                automatically if not specified.
-        :param experiment_name: (string) a name for the experiment, used for the save
-               directory
-        :param model_name: (string) a name for the model, used for the save
-               directory
-        :param model_resume_path: (string) path of a the model directory to
-               resume training of
+        :param dataset: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing the entire dataset to be used in the experiment.
+            If it has a split column, it will be used for splitting (0 for train,
+            1 for validation, 2 for test), otherwise the dataset will be
+            randomly split.
+        :param training_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing training data.
+        :param validation_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing validation data.
+        :param test_set: (Union[str, dict, pandas.DataFrame], default: `None`)
+            source containing test data.
+        :param training_set_metadata: (Union[str, dict], default: `None`)
+            metadata JSON file or loaded metadata.
+            Intermediate preprocess structure containing the mappings of the input
+            CSV created the first time a CSV file is used in the same
+            directory with the same name and a '.json' extension.
+        :param data_format: (str, default: `None`) format to interpret data sources. Will be inferred
+               automatically if not specified.
+        :param experiment_name: (str, default: `'experiment'`) name for
+            the experiment.
+        :param model_name: (str, default: `'run'`) name of the model that is
+            being used.
+        :param model_resume_path: (str, default: `None`) resumes training of
+            the model from the path specified. The difference with
+            model_load_path is that also training statistics like the current
```

Comment:
two times "also", the state of the optimizer is another thing that is resumed, should list it

Reply:
done